RunningOrder/Shared/AppStateIndicator.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  AppStateIndicator.swift
+//  RunningOrder
+//
+//  Created by Clément Nonn on 16/07/2021.
+//  Copyright © 2021 Worldline. All rights reserved.
+//
+
+import SwiftUI
+
+extension ErrorTrace: Identifiable {
+    var id: Date { self.date }
+}
+
+extension ErrorTrace: Equatable {
+    static func == (lhs: ErrorTrace, rhs: ErrorTrace) -> Bool {
+        lhs.date == rhs.date
+    }
+}
+
+struct AppStateIndicator: View {
+    @EnvironmentObject var appStateManager: AppStateManager
+
+    let ratio: CGFloat = 1/32
+    let size: CGFloat = 10
+
+    @State private var isTapped = false
+
+    var presentErrorList: Binding<Bool> {
+        Binding {
+            isTapped && !appStateManager.errors.isEmpty
+        } set: { value in
+            isTapped = value
+        }
+    }
+
+    var body: some View {
+        if appStateManager.currentLoading != nil {
+            ProgressView()
+                .progressViewStyle(CircularProgressViewStyle())
+                .scaleEffect(ratio * size)
+                .frame(width: size, height: size)
+        } else {
+            Circle()
+                .fill(appStateManager.errors.isEmpty ? Color.green : .red)
```

Comment:
juste .green here doesn't work ?

Reply:
no because fill doesn't take a Color object, but Shape object. Color conforms to Shape, but the autocomplete doesn't work, at least in the current version of Swift. (I think it will work in Xcode 13)